### PR TITLE
Add CRDs to etcd storage path test

### PR DESCRIPTION
**Pull request by enj**
This change updates the etcd storage path test to exercise custom resource storage by creating custom resource definitions before running the test.

Duplicated custom resource definition test logic was consolidated.

Signed-off-by: Monis Khan <mkhan@redhat.com>

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: test/integration/etcd/server.go [hunk 1]
```diff
@@ -281,3 +289,79 @@ func JSONToUnstructured(stub, namespace string, mapping *meta.RESTMapping, dynam
...
+
+func waitForEstablishedCRD(client apiextensionsclientset.Interface, name string) error {
+	return wait.PollImmediate(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+		crd, err := client.ApiextensionsV1beta1().CustomResourceDefinitions().Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range crd.Status.Conditions {
+			switch cond.Type {
+			case apiextensionsv1beta1.Established:
```
Comment (krmayankk) [inline test/integration/etcd/server.go]: what does Established mean ? Is this a new field we need to consider when using CRD's ?

Comment (liggitt) [reply test/integration/etcd/server.go]: That's a field in status you can monitor to see when the CRD has been accepted by the apiserver and surfaced as an actual resource. It's not new (was present in the first release of CRDs).

Comment (krmayankk) [reply test/integration/etcd/server.go]: So does this mean  under load or some corner conditions, this takes a non trivial amount of time , that we need to monitor this ? I always assumed once the  CRD was created , it was accepted ...

**Review by soltysh — APPROVED** [review]
/lgtm
